2-mandatory/3-financial-times.js

```diff
@@ -5,33 +5,58 @@
     Implement the function below, which will return a new array containing only article titles which will fit.
 */
 function potentialHeadlines(allArticleTitles) {
-    // TODO
+    const out = [];
+    for (z in allArticleTitles) {
+        if (ARTICLE_TITLES[z].length <= 65) {
+            out[z] = ARTICLE_TITLES[z];
+        }
+    }
+    return out
 }
 
 /*
     The editor of the FT likes short headlines with only a few words!
     Implement the function below, which returns the title with the fewest words.
     (you can assume words will always be seperated by a space)
 */
+function getNumWords(title) {
+    return title.split(' ').length;
+}
 function titleWithFewestWords(allArticleTitles) {
-    // TODO
+    if (allArticleTitles.length < 1) {
+        return '';
+    }
+    return allArticleTitles.reduce((shortestTitleSoFar, title) => {
+        return getNumWords(title) < getNumWords(shortestTitleSoFar)
+            ? title
+            : shortestTitleSoFar;
+    }, allArticleTitles[0]);
 }
 
 /*
     The editor of the FT has realised that headlines which have numbers in them get more clicks!
     Implement the function below to return a new array containing all the headlines which contain a number.
     (Hint: remember that you can also loop through the characters of a string if you need to)
 */
+titlesWithNumbers = []
 function headlinesWithNumbers(allArticleTitles) {
-    // TODO
+    allArticleTitles.forEach(all => {
```

Comment:
I think, 'all' is not good variable name we may use articleTitle, or title instead